packages/client/src/App.test.tsx

```diff
@@ -1,9 +1,75 @@
-import React from "react"
-import ReactDOM from "react-dom"
+import { ListItem } from "@material-ui/core"
+import { createHistory, createMemorySource } from "@reach/router"
+import { ReactWrapper } from "enzyme"
+import * as React from "react"
 import App from "./App"
+import Avatar from "./Avatar"
+import { mount, updates } from "./testing"
+import * as $ from "./testing/fixtures"
 
-it("renders without crashing", () => {
-  const div = document.createElement("div")
-  ReactDOM.render(<App />, div)
-  ReactDOM.unmountComponentAtNode(div)
+it("displays a list of conversations", async () => {
+  const app = mount(<App />, {
+    mocks: [$.getAccountMock],
+    route: "accounts/1/dashboard"
+  })
+  await updates(app)
+  expect(app.find(ListItem)).toIncludeText("Hello from test")
 })
+
+it("archives a conversation from the list view", async () => {
+  const app = mount(<App />, {
+    mocks: [
+      $.getAccountMock,
+      $.archiveMock,
+      {
+        ...$.getAccountMock,
+        result: { data: { account: { ...$.account, conversations: [] } } }
+      }
+    ],
+    route: `accounts/${$.account.id}/dashboard`
+  })
+  await updates(app)
+  const row = findConversationRow(app, $.conversation)
+  expect(row).toIncludeText("Hello from test")
+  row.find(Avatar).simulate("click")
+  app.find('button[aria-label="archive"]').simulate("click")
+  await updates(app, 10)
+  expect(findConversationRow(app, $.conversation)).not.toExist()
+})
+
+it("archives a conversation from the conversation view", async () => {
+  const history = createHistory(
+    createMemorySource(
+      `accounts/${$.account.id}/conversations/${$.conversation.id}`
+    )
+  )
+  const app = mount(<App />, {
+    mocks: [
+      $.getConversationMock,
+      $.setIsReadMock,
+      $.archiveMock,
+      {
+        ...$.getAccountMock,
+        result: { data: { account: { ...$.account, conversations: [] } } }
+      }
+    ],
+    history
+  })
+  await updates(app)
+  app.find("button[aria-label='archive']").simulate("click")
+  await updates(app, 10)
```

Comment:
Use the `updates` helper when you need to allow for a delay for the component to update after an asynchronous process, such as a GraphQL request. You can provide a second argument to specify a longer delay, given in milliseconds.